napari/utils/settings/_manager.py

```diff
@@ -0,0 +1,156 @@
+"""Settings management.
+"""
+
+import os
+from pathlib import Path
+
+from pydantic import ValidationError
+from yaml import safe_dump, safe_load
+
+from napari.utils.settings._defaults import CORE_SETTINGS
+
+
+class SettingsManager:
+    """
+    Napari settings manager using evented SettingsModels.
+
+    This provides the presistence layer for the application settings.
+
+    Parameters
+    ----------
+    config_path : str, optional
+        Provide the base folder to store napari configuration. Default is None,
+        which will point to `"~/.napari/settings/`.
+    save_to_disk : bool, optional
+        Persist settings on disk. Default is True.
+    """
+
+    _FILENAME = "settings.yaml"
+
+    def __init__(self, config_path: str = None, save_to_disk: bool = True):
+        # TODO: Use this instead Path.home() / ".config" / ".napari" ?
+        self._config_path = (
```

Comment:
can we also expose this as a read-only property?
```python
@property
def path(self):
	return self._config_path
```

Reply:
Fixing